### ISSUE TITLE
Panic when escaping values

### ISSUE THREAD

**Issue by epage** [tafia/quick-xml#87]
Steps to reproduce:
1. fetch my [xml branch of cobalt](https://github.com/epage/cobalt.rs/tree/xml).
2. `RUST_BACKTRACE=1 cargo test --test mod`

(sorry I've not taken time to dig into my output to RSS or RSS's output to quick-xml)

I'm making the assumption this is a quick-xml bug and not a rss bug because of my quick glance at the quick-xml code.  If this is wrong, I'll move this over to rss.

Backtrace
```
        thread 'rss' panicked at 'slice index starts at 28 but ends at 26', /checkout/src/libcore/slice/mod.rs:741:4
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
...
   8: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:92
   9: core::slice::slice_index_order_fail
             at /checkout/src/libcore/slice/mod.rs:741
  10: <core::ops::range::Range<usize> as core::slice::SliceIndex<[T]>>::index
             at /checkout/src/libcore/slice/mod.rs:864
  11: core::slice::<impl core::ops::index::Index<I> for [T]>::index
             at /checkout/src/libcore/slice/mod.rs:717
  12: quick_xml::escape::escape
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/quick-xml-0.9.2/src/escape.rs:57
  13: quick_xml::events::BytesText::escaped
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/quick-xml-0.9.2/src/events/mod.rs:344
  14: <quick_xml::writer::Writer<W>>::write_event
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/quick-xml-0.9.2/src/writer.rs:81
  15: <quick_xml::writer::Writer<W> as rss::toxml::WriterExt>::write_text_element
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/toxml.rs:59
  16: <rss::item::Item as rss::toxml::ToXml>::to_xml
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/item.rs:617
  17: <&'a T as rss::toxml::ToXml>::to_xml
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/toxml.rs:20
  18: <quick_xml::writer::Writer<W> as rss::toxml::WriterExt>::write_objects
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/toxml.rs:103
  19: <rss::channel::Channel as rss::toxml::ToXml>::to_xml
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/channel.rs:1284
  20: rss::channel::Channel::write_to
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/channel.rs:1048
  21: <rss::channel::Channel as alloc::string::ToString>::to_string
             at /home/epage/.cargo/registry/src/github.com-1ecc6299db9ec823/rss-1.0.0/src/channel.rs:1058
  22: cobalt::cobalt::create_rss
             at src/cobalt.rs:295
  23: cobalt::cobalt::build
             at src/cobalt.rs:197
...
```

**Comment by tafia** [comment]
Thanks for opening the issue. I'll try to have a look asap. Do you know which test in particular fails? It'll help finding the issue.

**Comment by epage** [comment]
Its cobalt's `rss` test in the "mod" collection of tests.

**Comment by tafia** [comment]
@epage I merged #88. Let me know if something is wrong with it.
Thanks!
